source/compiler-core/slang-source-loc.cpp

```diff
@@ -573,15 +573,38 @@ int SourceFile::calcColumnIndex(int lineIndex, int offset, int tabSize)
 
 void SourceFile::setContents(ISlangBlob* blob)
 {
-    const UInt contentSize = blob->getBufferSize();
+    const UInt rawContentSize = blob->getBufferSize();
 
-    SLANG_ASSERT(contentSize == m_contentSize);
+    SLANG_ASSERT(rawContentSize == m_contentSize);
 
-    char const* contentBegin = (char const*)blob->getBufferPointer();
-    char const* contentEnd = contentBegin + contentSize;
+    Byte* rawContentBegin = (Byte*)blob->getBufferPointer();
 
-    m_contentBlob = blob;
-    m_content = UnownedStringSlice(contentBegin, contentEnd);
+    // Query the encoding type and discard the Unicode Byte-Order-Marker before decoding
+    size_t offset;
+    auto type = CharEncoding::determineEncoding(
+        rawContentBegin,
+        rawContentSize,
+        offset);
+    SLANG_ASSERT(rawContentSize >= offset);
+
+    List<char> decodedBuffer;
+    CharEncoding::getEncoding(type)->decode(
+        rawContentBegin + offset,
+        rawContentSize - offset,
+        decodedBuffer);
+    List<uint8_t> decodedBuffer2;
```

Comment:
It is unfortunate that we need to do a full copy just to convert List<char> to List<uint8_t>. We can probably implement some Move operator on a List to allow us to move the buffer pointer without a copy, admittedly this is likely going to be a UB.

Reply:
Yes, I agree is unfortunate and I imagined there must be some better solution, but it seems to be moot as Jay proposed using String + StringBlob.  Though it still requires the copy, since it's from a List to a String, it's doesn't seem as unnecessary.